mp4parse/src/lib.rs

```diff
@@ -1412,6 +1413,7 @@ fn read_ds_descriptor(data: &[u8], esds: &mut ES_Descriptor) -> Result<()> {
     esds.audio_object_type = Some(audio_object_type);
     esds.audio_sample_rate = sample_frequency;
     esds.audio_channel_count = Some(channel_counts);
+    esds.decoder_specific_data.extend(data.iter());
```

Comment:
You're setting the entire contents of data here, not adding to it, right? Is it worth asserting that decoder_specific_data is empty before calling extend?